src/game/common/damagefx.cpp

```diff
@@ -13,7 +13,176 @@
  *            LICENSE
  */
 #include "damagefx.h"
+#include "fxlist.h"
 
 #ifndef GAME_DLL
 DamageFXStore *g_theDamageFXStore;
 #endif
+
+const FieldParse *DamageFX::Get_Field_Parse()
+{
+    static FieldParse myFieldParse[] = { { "AmountForMajorFX", &DamageFX::Parse_Amount, nullptr, 0 },
+        { "MajorFX", &DamageFX::Parse_Major_FX_List, nullptr, 0 },
+        { "MinorFX", &DamageFX::Parse_Minor_FX_List, nullptr, 0 },
+        { "ThrottleTime", &DamageFX::Parse_Time, nullptr, 0 },
+        { "VeterancyAmountForMajorFX", &DamageFX::Parse_Amount, g_veterancyNames, 0 },
+        { "VeterancyMajorFX", &DamageFX::Parse_Major_FX_List, g_veterancyNames, 0 },
+        { "VeterancyMinorFX", &DamageFX::Parse_Minor_FX_List, g_veterancyNames, 0 },
+        { "VeterancyThrottleTime", &DamageFX::Parse_Time, g_veterancyNames, 0 },
+        { nullptr, nullptr, nullptr, 0 } };
+
+    return myFieldParse;
+}
+
+void Parse_Common_Stuff(INI *ini,
+    const char *const *names,
+    VeterancyLevel &min_veterancy_level,
+    VeterancyLevel &max_veterancy_level,
+    DamageType &min_damage_type,
+    DamageType &max_damage_type)
+{
+    if (names != nullptr) {
+        min_veterancy_level = static_cast<VeterancyLevel>(ini->Scan_IndexList(ini->Get_Next_Token(), names));
+        max_veterancy_level = min_veterancy_level;
+    } else {
+        min_veterancy_level = VETERANCY_REGULAR;
+        max_veterancy_level = VETERANCY_HEROIC;
+    }
+
+    const char *damage = ini->Get_Next_Token();
+
+    if (strcasecmp(damage, "Default") == 0) {
+        min_damage_type = DAMAGE_EXPLOSION;
+        max_damage_type = DAMAGE_STATUS;
```

Comment:
I suggest to write `DAMAGE_NUM_TYPES - 1` here to make it more robust against potential future custom damage type changes.

Reply:
fixed.